### PR TITLE
Setup-associations📖📖🏛️🏛️👨‍🎓👨‍🎓

### PR DESCRIPTION
**Pull request by AnsarIbrahim**
# This PR Contains:

## Project requirements:

- Create a class Classroom✅
- Create the has-many/belongs-to relationship between Classroom and Student✅
- Create a class Book✅
- Create a class Rental✅
- Create the many-to-many relationship between Person and Book using the intermediate class Rental✅

### CODE REVIEW

File: school_library/setup/book.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Book
+  attr_accessor :title, :author
```
Comment (franclobo) [inline school_library/setup/book.rb]: - [ ] Kindly add :rentals to attr_accesor, getters and Setters for @rentals should be present.

File: school_library/setup/classroom.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label, :students
```
Comment (franclobo) [inline school_library/setup/classroom.rb]: - [ ] Kindly add `:students` to attr_reader, just getters for @students should be present.

File: school_library/setup/student.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Student
+  attr_accessor :name, :classroom
```
Comment (franclobo) [inline school_library/setup/student.rb]: - [ ] Kindly add `classroom` to attr_reader, just getters for @classroom should be present.

File: school_library/setup/student.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class Student
```
Comment (franclobo) [inline school_library/setup/student.rb]: - [ ] Please 🙏🏽 class Student should inherit from class Person to be able to use methods from other class.

File: school_library/setup/student.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  attr_accessor :name, :classroom
+
+  def initialize(name)
+    @name = name
+    @classroom = nil
+    @rentals = []
+  end
+
+  attr_reader :rentals
```
Comment (franclobo) [inline school_library/setup/student.rb]: - [ ] Please 🙏🏽 add a method that serves as the 'belongs-to' side (a student belongs to a classroom). Link to the article with [examples](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)

File: school_library/setup/student.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+class Student
+  attr_accessor :name, :classroom
+
+  def initialize(name)
```
Comment (franclobo) [inline school_library/setup/student.rb]: - [ ] Kindly initialize your method with all the parameters required to create a student those are name, age, classroom, and parent_permision. Additionally, we should use the super keyword super is used to call the constructor (initializer) of the parent class. This is a common practice in object-oriented programming, where a child class can reuse and extend the behavior of its parent class's constructor. 👇🏽 

```ruby
def initialize(age, classroom, name = 'Unknow', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
```

File: school_library/setup/book.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (franclobo) [inline school_library/setup/book.rb:8]: - [ ] Please 🙏🏽 add a method for adding rentals. Link to the [example](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship). The book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @AnsarIbrahim, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [x] Due to it is not possible to modify the commits, please 🙏🏽 consider do not use too much emojis on your commits and branch's name in the future for professional documentation. 
- [x]  Please 🙏🏽 add the association in the person class in order to achieve the the 'has-many' association between books and person through rentals.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rotimiazeez — APPROVED** [review]
Hi @AnsarIbrahim 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job working on the required changes 👌🏽
- All associations are well set 👍🏽
- Good PR description and title 👏🏽

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**